security/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,281 @@
+
+### FM5: BLS Precompiles could cause increased FP Program execution time
+
+- **Description:** If implemented in the FP program, BLS precompiles could greatly increase computation time.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+	1. Fully mitigated by using an accelerated precompile that calls out to L1 instead of calculating inside the program.
+        - [Specs](https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/isthmus/exec-engine.md#evm-changes)
+- **Detection:** The `vm-runner` computation time would **increase** significantly, or **panic** due to OOM and trigger an alert through our current set of monitoring. 
+- **Recovery Path(s)**: If this occurred, which it can't with the mitigations, we'd have to make op-program efficient enough to execute
+BLS precompiles in sufficient time.
+
+### FM6: BLS Precompiles could fail to execute in the FP program
+
+- **Description:** If implemented in the FP program, BLS precompiles could fail to execute. This could cause certain blocks to be unprovable.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+	1. Fully mitigated by using an accelerated precompile that calls out to L1 instead of calculating inside the program. This means it should 
+  have the same performance characteristics as existing crypto precompiles like `ecrecover`.
+        - [Specs](https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/isthmus/exec-engine.md#evm-changes)
+- **Detection:** op-program trace runner failure (possibly triggering an alert)
+- **Recovery Path(s)**: If this occurred, which it can't with the mitigations, we'd have to make op-program efficient enough to execute BLS precompiles in sufficient time.
+
+### FM7: Early fork if batches containing EIP-7702 transactions could be posted before Pectra
+
+- **Description:** If batches containing EIP-7702 transactions are posted and accepted early, the batcher could fork any nodes that upgrade early. A malicious batcher could cause this to happen if we don't have a check to ensure type 4 (SetCode) txs can only appear in a batch after Isthmus is active.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+  1. Spec disallows type 4 transactions before Isthmus
+	1. End-to-end test
+        - [Action test created here](https://github.com/ethereum-optimism/optimism/blob/01ddb2e6a09edf55a7cb2130e0a5b6acd0c2d2fa/op-e2e/actions/upgrades/isthmus_fork_test.go#L290)
+- **Detection:** Upgraded clients would fork causing an alert on L2 unsafe liveness.
+- **Recovery Path(s)**: Emergency patch to rollup node clients (like `op-node`).
+
+
+### FM8: Smart contracts relying on sender check no longer ensures sender does not have code
+
+- **Description:** As part of EIP-7702, `msg.sender == tx.origin` no longer implies that the caller has no code. If unaddressed, this could cause
```

Comment:
This has been handled, there is a new library for handling this check that has been introduced into our L1 contracts

Reply:
Correct the changes has been made on L1. 
Also this is fixed on the L2 the modifier `onlyEOA` is used now: 
https://github.com/ethereum-optimism/optimism/blob/9249efc6343208f69283290fc9c5c8f6e7b243f8/packages/contracts-bedrock/src/L2/L2StandardBridge.sol#L104

I would like still to know if there is any operation on the `op-node` that can potentially revert or cause unexpected behavior if interacting with an EIP7702 address such as minting, refund etc. 
But the smart contractsa are correct.

Reply:
> we should be safe from eip-3607 from just rebasing on top of latest geth, but good call on checking that deposits work as expected, https://eips.ethereum.org/EIPS/eip-3607

From Mark, this is correct but just in case can someone point me the tests that can cover this if there is?